apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -257,6 +288,19 @@ public override async Task AcknowledgeAsync(SubscriptionName subscription, IEnum
                 {
                     await _taskHelper.ConfigureAwait(_scheduler.Delay(_writeAsyncPreDelay, CancellationToken.None));
                 }
+
+                // Simulate Ack failure if _ackModifyAckDeadlineAction is specified.
+                // All gRPC exceptions from this method are fire and forget, so no exception should propogate to the caller.
+                // Non-gRPC exceptions will propogate to the caller.
+                if (_ackModifyAckDeadlineAction?.Exception != null && _ackModifyAckDeadlineAction.OnAck)
+                {
+                    if (Interlocked.CompareExchange(ref _numberOfAckModifyAckDeadlineFailures, _ackModifyAckDeadlineAction.NumberOfFailures, _ackModifyAckDeadlineAction.NumberOfFailures) < _ackModifyAckDeadlineAction.NumberOfFailures)
+                    {
+                        Interlocked.Increment(ref _numberOfAckModifyAckDeadlineFailures);
+                        await Task.FromException(_ackModifyAckDeadlineAction.Exception);
```

Comment:
Why are you awaiting something you know will throw, rather than just using `throw _ackModifyAckDeadlineAction.Exception`?

Reply:
I was somehow so keen to use `Task.FromException` that `throw` didn't come to my mind. I have now changed it to throw.